src/sentry/relay/config/metric_extraction.py

```diff
@@ -533,26 +522,30 @@ def _convert_aggregate_and_query_to_metrics(
         if not should_use_on_demand_metrics(dataset, aggregate, query, groupbys, prefilling):
             return None
 
-        on_demand_spec = OnDemandMetricSpec(
-            field=aggregate,
-            query=query,
-            environment=environment,
-            groupbys=groupbys,
-            spec_type=spec_type,
-            use_updated_env_logic=use_updated_env_logic,
-        )
-
-        metric_spec = on_demand_spec.to_metric_spec(project)
-        # TODO: switch to validate_rule_condition
-        if (condition := metric_spec.get("condition")) is not None:
-            validate_sampling_condition(json.dumps(condition))
-        else:
-            metrics.incr(
-                "on_demand_metrics.missing_condition_spec",
-                tags={"prefilling": prefilling},
+        # XXX: We could remove deduplication here
```

Comment:
I could handle deduplication here in case we want to still report duplicates as errors in `_merge_metric_specs`.